eth/catalyst/api.go

```diff
@@ -435,8 +435,17 @@ func (api *ConsensusAPI) forkchoiceUpdated(update engine.ForkchoiceStateV1, payl
 	// sealed by the beacon client. The payload will be requested later, and we
 	// will replace it arbitrarily many times in between.
 	if payloadAttributes != nil {
-		if api.eth.BlockChain().Config().Optimism != nil && payloadAttributes.GasLimit == nil {
-			return engine.STATUS_INVALID, engine.InvalidPayloadAttributes.With(errors.New("gasLimit parameter is required"))
+		if api.eth.BlockChain().Config().Optimism != nil {
+			if payloadAttributes.GasLimit == nil {
+				return engine.STATUS_INVALID, engine.InvalidPayloadAttributes.With(errors.New("gasLimit parameter is required"))
+			}
+			if api.eth.BlockChain().Config().IsHolocene(payloadAttributes.Timestamp) {
+				if payloadAttributes.EIP1559Params == nil {
```

Comment:
I believe we need the validity check on it not being 0 here? https://github.com/ethereum-optimism/specs/pull/421

Reply:
Thanks, I was assuming the check I put in BuildPayload would have triggered this, but I guess that can happen async. fixed.